driver-core/src/main/com/mongodb/internal/operation/CommandOperationHelper.java

```diff
@@ -227,42 +206,44 @@ public T call(final ConnectionSource source) {
 
             if (!shouldAttemptToRetryRead(retryReads, e)) {
                 if (retryReads) {
-                    logUnableToRetry(command.getFirstKey(), e);
+                    logUnableToRetry(Objects.requireNonNull(command).getFirstKey(), e);
```

Comment:
If command is null then both `command.getFirstKey()` and `Objects.requireNonNull(command).getFirstKey()` act the same.  Which is probably isn't desired, I'd imagine if command is null then its better to rethrow the original exception rather than a NPE.

Reply:
Good catch.  I blindly followed IntelliJ's advice here instead of seeing what the real problem was.  There is no way that command should ever be null, and if it is we should certainly not retry. Pulling the initialization of command outside the try block gets rids of the warning.

Reply:
Done